### PR TITLE
toolchains: Fix gcc linker fail

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

### CODE REVIEW

File: toolchains/rbe_toolchains_config.bzl [hunk 1]
```diff
@@ -36,7 +36,7 @@ _CLANG_LIBCXX_ENV = dicts.add(_CLANG_ENV, {
 _GCC_ENV = {
     "BAZEL_COMPILER": "gcc",
     "BAZEL_LINKLIBS": "-l%:libstdc++.a",
-    "BAZEL_LINKOPTS": "-lm",
+    "BAZEL_LINKOPTS": "-lm:-fuse-ld=gold",
```
Comment (phlax) [inline toolchains/rbe_toolchains_config.bzl:39]: this seems obsolete - it didnt seem to make any difference when regenerating chains

File: toolchains/gcc.env.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "BAZEL_COMPILER": "gcc",
   "BAZEL_LINKLIBS": "-l%:libstdc++.a",
-  "BAZEL_LINKOPTS": "-lm",
+  "BAZEL_LINKOPTS": "-lm:-fuse-ld=gold",
```
Comment (phlax) [inline toolchains/gcc.env.json:4]: not sure if this is correct, but it reverts it to what it was using previously

**Comment by phlax** [comment]
cc @lizan

**Comment by phlax** [comment]
for context the problem this addresses can be seen here:

https://dev.azure.com/cncf/envoy/_build/results?buildId=100138&view=logs&j=8bf29878-a4cc-50f7-4e84-2255e6fd4065&t=3267cd16-a387-57e0-0ef4-d7a4d1c6c089&l=97

**Comment by lizan** [comment]
I'm looking into this

File: toolchains/gcc.env.json (hunk 2 above)
Comment (lizan) [reply toolchains/gcc.env.json:4]: I think this one is correct

**Review by lizan — APPROVED** [review]

**Pull request merged**